openrtb_ext/request_wrapper.go

```diff
@@ -1278,8 +1279,10 @@ func (re *RegExt) unmarshal(extJson json.RawMessage) error {
 
 	gpcJson, hasGPC := re.ext[gpcKey]
 	if hasGPC && gpcJson != nil {
-		if err := jsonutil.Unmarshal(gpcJson, &re.gpc); err != nil {
-			return err
+		gpcResult := gjson.ParseBytes(gpcJson)
+		if gpcResult.Exists() {
+			gpc := gpcResult.String()
+			re.gpc = &gpc
```

Comment:
I saw that gjson was already being used, and I know that it's much more lenient. Are you suggesting that we roll this fix into the jsonutil library somehow? like a `ReadString` method or something to that effect (that could be the thing using gjson)

- 51 Degrees already "broke that seal" #3893

Reply:
I moved this into a new jsonutil function `ParseIntoString` LMK if you are ok with that name; technically it parses into a string pointer…

Reply:
@scr-oath we discussed offline and are good with `gjson`. We discussed alternatives like using `jsonparser` which is used extensively throughout the code base but we could eventually move away from or changing the type of `gpc` to some custom type like `util/jsonutil/StringInt` which has an unmarshal method override.

I like that you moved this to a jsonutil function.@scr-oath we discussed offline and are good with `gjson`. We discussed alternatives like using `jsonparser` which is used extensively throughout the code base but we could eventually move away from or changing the type of `gpc` to some custom type like `util/jsonutil/StringInt` which has an unmarshal method override.

I like that you moved this to a jsonutil function.

Reply:
gjson is actually quite fast… we use it a lot for the json.RawMessage stuff because just looking for a value is much faster than unmarshalling.

What I also love about it is that you can read json and then interrogate its structure.  We actually use it in our slog to OpenTelemetry Handler to convert into rich, recursive otel types (primitive, map, array) that correspond to the json counterparts.